### ISSUE TITLE
GetxController are not mockable

### ISSUE THREAD

**Issue by danilocoppi** [jonataslaw/getx#533]
**Describe the bug**
On tests when you try to mock a GetxController and then try to put it on Get.put it throws a null call exception.

**To Reproduce**
Run a flutter test

**Expected behavior**
Should work fine. We wer using version 3.4.x and it was working fine.
Now on 3.6.1 it does not work anymore.

**Trace Exception**
The method 'call' was called on null.
Receiver: null
Tried calling: call()
dart:core                                                                   Object.noSuchMethod
package:get/src/instance/get_instance.dart 178:16                           GetInstance._startController
package:get/src/instance/get_instance.dart 152:7                            GetInstance._initDependencies
package:get/src/instance/get_instance.dart 219:7                            GetInstance.find
package:get/src/instance/get_instance.dart 86:12                            GetInstance.put
package:get/src/instance/extension_instance.dart 72:12                      Inst.put
test/main_getx_test.dart 5:7                                                main
===== asynchronous gap ===========================
package:test_api                                                            RemoteListener.start
/tmp/flutter_tools.DJSWUH/flutter_test_listener.IZBBRF/listener.dart 16:25  serializeSuite
/tmp/flutter_tools.DJSWUH/flutter_test_listener.IZBBRF/listener.dart 42:36  main

**Flutter Version:**
Flutter 1.20.2

**Get Version:**
GetX 3.6.1

**Describe on which device you found the bug:**
On Testing - flutter test ...

**Minimal reproduce code**
```dart
import 'package:get/get.dart';
import 'package:mockito/mockito.dart';

void main() async {
  Get.put<Sample>(MockSample());
  print('done');
}

class Sample extends GetxController {}

class MockSample extends Mock implements Sample {}
```

**Comment by danilocoppi** [comment]
Here is the fix
You need to check if onStart exists on get_instance.dart
(line 174)

```dart
  void _startController<S>({String tag}) {
    final key = _getKey(S, tag);
    final i = _singl[key].getDependency();
    if (i is DisposableInterface) {
      if (i.onStart != null) i.onStart();
      GetConfig.log('"$key" has been initialized');
    }
  }
``

**Comment by jonataslaw** [comment]
Thanks I'll update now

**Comment by jonataslaw** [comment]
Fix on 3.7
